### DISCUSSION TITLE
Is cache-hit also true when restore-keys is hit?

### DISCUSSION THREAD

**Discussion by sebastianrath** [actions/cache#789] [Q&A] [answered]
Hi, I am trying to find information about `steps.cache.outputs.cache-hit`. This is the snippet of my workflow

```
        uses: actions/cache@v2
        id: cache
        with:
          path: |
            **/node_modules
            **/.eslintcache
          key: ${{ runner.os }}-benchmark-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-benchmark
            ${{ runner.os }}-
```

As you can see, I use `restore-keys` as a fallback. I would like to make `npm ci` dependent on a cache-hit. This would only makes sense if `cache-hit` is false despite a potential cache-hit through `restore-keys`.

```
      - name: 📥 Install dependencies
        if: steps.cache.outputs.cache-hit != 'true'
        run: npm ci
```

Any help is highly appreciated!

**Comment by jongwooo** [comment, marked as answer]
If there is a cache that fits `key`, it is called `cache-hit`, and the action restores the cache file to `path`. If the cache does not match `key`, it is called `cache-miss`, and a new cache is created when the Job is successfully terminated. When `cache-miss` occurs, the action uses `restore-keys` as an alternative key to rediscover. In this case, `cache-hit` returns false.
